src/core/handle.ml

```diff
@@ -186,7 +188,48 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
       out 3 "(symb) %s ≔ %a\n" x.elt pp_term t;
       let d = if op then None else Some(t) in
       let ss = Sig_state.add_symbol ss e Defin x a impl d in
-      (ss, None)
+      (ss, None, None)
+  | P_inductive(e, s, t, tl) ->
+      (* Obtaining of the global type *)
+      let type_symbol_head = match t with
+        | None -> { elt = P_Type ; pos = None }
+        | Some a -> a
+      in
+      (* Add the first symbol in the signature *)
+      let (ss, _, _) = handle_cmd ss
+                      ({elt = P_symbol(e, Defin, s, [], type_symbol_head);
+                        pos = cmd.pos})
+      in
+      let create_symbol : string -> term -> term option -> sym  =
```

Comment:
This function doesn't seem necessary as this is done in handle_symbol.

Reply:
You just need to modify Sig_state.add_symbol and handle_symbol so that they return the added symbol as well.

Reply:
Maybe the name of the function is a bit confusing.
The function `create_symbol` is useful to build my record, and now I use `handle_symbol` (and not `handle_cmd`) to add the symbols in the signature.

Reply:
Hi @amelieled . Could you please make handle_symbol return a symbol. Then, you do not need to change the type of handle_cmd and make it recursive? You just have to call handle_symbol from handle_inductive. Thanks.